mmv1/third_party/terraform/services/compute/resource_compute_region_instance_group_manager.go.tmpl

```diff
@@ -1145,6 +1210,44 @@ func flattenRegionUpdatePolicy(updatePolicy *compute.InstanceGroupManagerUpdateP
 	return results
 }
 
+{{ if ne $.TargetVersionName `ga` -}}
+func expandInstanceFlexibilityPolicy(d *schema.ResourceData) *compute.InstanceGroupManagerInstanceFlexibilityPolicy {
```

Comment:
Why do we need to check HasChange here instead of just Get? Can we not just send the final value for the field to the API?

Reply:
This expand method is used both for create and update. For create sending the final value to the api is fine. But IIUC, we use patch api to update IGM trough terraform where we only select the changes to update it. Also the first check about "d.HasChange("instance_flexibility_policy")", right now we have only instanceSelections within flexibilityPolicy, but later we will add some other fields (e.g provisioningModelMix) to extend this resource. So for patch update we should check about any changes of the field and then need to expand. Isn't it?

Reply:
Oh OK- this is another case of GCE having nonstandard patch where individual elements need to have explicit nulls sent to be removed. Thanks.

Specifically, GCE adopted JSON patch as a GCE-only standard. In most APIs patch calls are an implicit mask over PUT-like semantics, where the supplied values for a list are assumed to be authoritative.

This means that sending the message `[]` is a noop in GCE and removes all array entries for most GCP APIs.

Reply:
Thanks for details explanation. 
Btw, I am in a little bit hesitate about the check "d.HasChange("instance_flexibility_policy")", obviously not the logic. We already have this check on the caller of update method. So for update we actually don't need to check it again here, I think. So I just removed this check from the beginning of the method.  Not sure if it matters somehow for create method specially about null check exceptions. Though so far the test are passing with or without this new resource field. 
Check please.

Reply:
Should be fine to remove.